### PR TITLE
Add the functionality to allow users to withdraw any amount from the vault

### LINKED ISSUE #10: Add the function to allow users withdraw a specific amount from the vaults

### PR DESCRIPTION
**Pull request by arjanvillon**
# Project: New Order DAO App Frontend

## Motivation and Context

Why is this change required? What problem does it solve?
- Need a way to allow the users to withdraw a certain amount

## Description

Describe your changes in detail.
- Use the withdraw function from the contracts
- Update action buttons

### Related Issues

Resolves #10 

### How Has This Been Tested?

Please describe in detail how you tested your changes.
- Used the development contracts to run the functions
- Withdraw a small amount like `10000` and check the updated balance
- Withdraw Maximum amount and should still be left with rewards

### Screenshots/Video (if appropriate):

If this PR covers a UI or workflow change, an image displaying the change should accompany the PR

image

_**After clicking the withdraw button**_

image

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by justingreenberg — APPROVED** [review]
LGTM 👍

File: src/components/Vaults/SLPVault.tsx [hunk 1]
```diff
@@ -250,4 +250,31 @@ export default class SLPVault {
...
+      const withdrawAmount = new BigNumber(amount);
+
+      const estimate = await this.slpVaultInstance.methods
+        .withdraw(withdrawAmount.shiftedBy(18))
+        .estimateGas({ from: who });
+
+      const tx = await this.slpVaultInstance.methods
+        .withdraw(withdrawAmount.shiftedBy(18))
+        .send(
+          { from: who, gas: Math.round(estimate * 1.2) },
```
Comment (brokenomics) [inline src/components/Vaults/SLPVault.tsx:265]: Where does the 1.2 come from?

File: src/pages/InvestPage/components/ActiveDepositsTab/components/Deposits/Deposits.tsx [hunk 2]
```diff
@@ -42,12 +46,189 @@ export const Deposits: React.FC<ActiveDepositsProps> = () => {
...
+    const tx = address && (await vaultClient.withdraw(address, newoWithdraw));
+
+    setPending(false);
+
+    if (tx) {
+      if (updateTokenBalance) await updateTokenBalance();
+
+      showNotification({
+        type: NOTIFICATION_TYPES.SUCCESS,
+        description: 'Withdrawing assets successfully',
```
Comment (brokenomics) [inline src/pages/InvestPage/components/ActiveDepositsTab/components/Deposits/Deposits.tsx:77]: nit: "Withdrew assets successfully"

File: src/pages/InvestPage/components/ActiveDepositsTab/components/Deposits/Deposits.tsx [hunk 3]
```diff
@@ -42,12 +46,189 @@ export const Deposits: React.FC<ActiveDepositsProps> = () => {
...
+            >
+              Withdraw
+            </Button>
+          </div>
+        </div>
+      </LoadingOverlay>
+    </Modal>
+  ));
+
+  //
```
Comment (brokenomics) [inline src/pages/InvestPage/components/ActiveDepositsTab/components/Deposits/Deposits.tsx:145]: forgot to write this comment? :kek:

**Review by brokenomics — COMMENTED** [review]
Overall LGTM, couple nits and a question.

File: src/components/Vaults/SLPVault.tsx (hunk 1 above)
Comment (justingreenberg) [reply src/components/Vaults/SLPVault.tsx:265]: i think i remember settling on this number after a conversation w/ @marek-new, a lot of the tx failures specifically for vaults were related to low gas... initially we thought updating the web3 library to latest fixed it, but people have still been having the same problem, so i believe we picked 1.2 kinda arbitrarily to boost the gas a bit

**Pull request merged**